bin/grouper-api

```diff
@@ -29,6 +31,66 @@ class Application(tornado.web.Application):
         super(Application, self).__init__(*args, **kwargs)
 
 
+class SendEmailThread(Thread):
+    """Background thread for sending emails from the async pool
+
+    This class thread will exist on multiple servers in a standard Grouper production environment
+    so we need to ensure that it's race-safe.
+    """
+
+    def send_emails(self, session, now_ts):
```

Comment:
This goes part of the way to mocking up the state for testing.  I think we want a dry_run parameter as well so that we can test the notification system.